### PR TITLE
DG and UG Changes

### PR DESCRIPTION
**Pull request by Aak242**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -256,8 +262,8 @@ The `updateManager` method facilitates the updating of tasks. It prompts the use
 #### Method Signature
 
 ```
-public void updateManager(Scanner scanner, WeekView weekView, MonthView monthView, boolean inMonthView,
-                          TaskManager taskManager, int day, int taskIndex, String newDescription)
+public void updateManager(Scanner scanner, WeekView weekView, MonthView monthView, boolean inMonthView, 
+        TaskManager taskManager, int day, int taskIndex, String newDescription)
```
Comment (clarencepohh) [inline docs/DeveloperGuide.md:266]: Good catch on the indentation mistakes!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -287,6 +293,8 @@ public void updateManager(Scanner scanner, WeekView weekView, MonthView monthVie
 
 ### Update Task Method
 
+![UpdateTaskSequenceDiagram.png](images%2Fsequence%2FUpdateTaskSequenceDiagram.png)
```
Comment (clarencepohh) [inline docs/DeveloperGuide.md:296]: Is the image path specified correctly?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -440,6 +448,95 @@ public static void deleteAllTasksOnDate (TaskManager taskManager, LocalDate spec
 - Gets the LocalDate requested by the caller.
 - Calls the `deleteTask` method on all tasks on the date.
 
+<a name="managing-free-time"></a>
+
+### Managing Free Time
+
+<a name="free-times-manager-method"></a>
```
Comment (clarencepohh) [inline docs/DeveloperGuide.md:455]: Good use of anchors! However, I think Intellij IDE can generate these for you!

File: docs/diagrams/class/Exceptions.puml [hunk 4]
```diff
@@ -2,48 +2,55 @@
...
+
++ TaskManagerException(errorMessage : String)
+
+{static} + checkIfDateTimeInFormat(dateTime : String) : void
+{static} + checkIfDateInFormat(date : String) : void
+{static} + checkIfTimeInFormat(time : String) : void
+{static} + checkIfDateInCurrentWeek(date : LocalDate, weekView : WeekView) : void
+{static} + checkIfTaskExistsInCurrentDate(dayTasks: List<Task>, index: int) : void
+{static} + checkIfDateInCurrentMonth(date : LocalDate) : void
+{static} + checkIfDateHasTasks(dayTasks : List<Task>) : void
```
Comment (clarencepohh) [inline docs/diagrams/class/Exceptions.puml:20]: Nice fixes to static variables as requested!

File: docs/diagrams/class/Exceptions.puml
Comment (clarencepohh) [inline docs/diagrams/class/Exceptions.puml:1]: Thanks for adding UML diagrams for Exceptions!

File: docs/images/sequence/UpdateTaskSequenceDiagram.png
Comment (clarencepohh) [inline docs/images/sequence/UpdateTaskSequenceDiagram.png:1]: Looks very small / hard to read. Possible to scale it up / remove unneeded methods (omit them on purpose)?

**Review by clarencepohh — CHANGES_REQUESTED** [review]
Might need some fixes to UML and image paths.
Otherwise great job settling the issues!

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -440,6 +448,95 @@ public static void deleteAllTasksOnDate (TaskManager taskManager, LocalDate spec
...
+
+### Free Times Manager Method
```
Comment (kyhjonathan) [inline docs/DeveloperGuide.md:457]: Good job on documentation of freeTimesManager!

File: docs/diagrams/sequence/UpdateTaskSequenceDiagram.puml [hunk 6]
```diff
@@ -0,0 +1,91 @@
+@startuml
```
Comment (kyhjonathan) [inline docs/diagrams/sequence/UpdateTaskSequenceDiagram.puml:1]: Sequence Diagrams LGTM!

File: docs/diagrams/class/Exceptions.puml (hunk 4 above)
Comment (NicholasTanYY) [inline docs/diagrams/class/Exceptions.puml:20]: Good job on making these static!

File: src/main/java/data/TaskManager.java [hunk 7]
```diff
@@ -718,26 +719,6 @@ public void addTasksFromFile(Map<LocalDate, List<Task>> tasksFromFile) throws Ta
...
-            throws TaskManagerException {
-        List<Task> dayTasks = taskManager.getTasksForDate(date);
-        checkIfDateHasTasks(dayTasks);
-
-        System.out.println(message);
-        for (int i = 0; i < dayTasks.size(); i++) {
-            System.out.println((i + 1) + ". " + dayTasks.get(i).getName());
-        }
-    }
-
```
Comment (NicholasTanYY) [inline src/main/java/data/TaskManager.java:740]: Good thinking, moving this code chunk to an unused section.

File: src/main/java/data/TaskManager.java [hunk 8]
```diff
@@ -858,7 +839,7 @@ public List<String> getFreeTimeSlots(List<Task> events, LocalDate currentDate) {
 
         // Add remaining time of the day to free slots only if lastEndTime is not endOfDay
         if (!lastEndTime.equals(endOfDay) && Duration.between(lastEndTime, endOfDay).toMinutes() > 0) {
-            freeTimeSlots.add(lastEndTime.toString() + " - " + endOfDay.toString());
+            freeTimeSlots.add(lastEndTime + " - " + endOfDay.toString());
```
Comment (NicholasTanYY) [inline src/main/java/data/TaskManager.java:842]: Good catch on fixing this bug!

File: src/main/java/commandparser/StringParser.java [hunk 9]
```diff
@@ -61,7 +61,7 @@ public static String parsePriorityLevel(String priorityString) throws TaskManage
         if (trimmedPriority.equals("H") || trimmedPriority.equals("M") || trimmedPriority.equals("L")) {
             return trimmedPriority;
         } else {
-            throw new TaskManagerException("Invalid priority level. Please use 'high', 'medium', or 'low'.");
+            throw new TaskManagerException("Invalid priority level. Please use 'H', 'M', or 'L'.");
```
Comment (clarencepohh) [inline src/main/java/commandparser/StringParser.java:64]: Good catch on outdated code!

File: src/main/java/data/exceptions/TaskManagerException.java [hunk 10]
```diff
@@ -14,10 +12,6 @@ public class TaskManagerException extends Exception {
             "The date must be within the current week. Please try again.";
     public static final String NOT_CURRENT_MONTH_MESSAGE =
             "The date must be within the current month. Please try again.";
-    public static final String INVALID_DATE_FORMAT_MESSAGE =
-            "Invalid date format. Please enter a valid date in the format: dd/MM/yyyy";
-    public static final String INVALID_TIME_FORMAT_MESSAGE =
-            "Invalid time format. Please enter a valid time in the format: HH:mm";
```
Comment (NicholasTanYY) [inline src/main/java/data/exceptions/TaskManagerException.java:20]: Thank you for removing the unused lines of code!

**Review by clarencepohh — APPROVED** [review]
Looks good overall! Thanks for making the changes requested.

**Review by NicholasTanYY — APPROVED** [review]
LGTM. Thank you for making changes to the code to fix final bugs!

**Pull request closed without merging**